### PR TITLE
FEATURE: Server getlegendgraphics json rule

### PR DESCRIPTION
**Pull request by elpaso**
Add a server WMS vendor parameter `SHOWRULEDETAILS` (bool), when it is passed to a GetLegendGraphic request with JSON FORMAT adds the rule expression to the legend item entry.

For instance:
GET /?SERVICE=WMS&VERSION=1.30&REQUEST=GetLegendGraphic&LAYERS=layer1&FORMAT=application/json&SHOWRULEDETAILS=1

Returns a JSON with a `rule` field added
```json
{"nodes":[{
"symbols":[{
"icon":"...",
"title":"Detaljplaner","rule": "plantyp = '1'"}

```

Funded by: Kristianstadt

### CODE REVIEW

File: tests/src/python/test_qgsserver_wms_getlegendgraphic.py [hunk 1]
```diff
@@ -1360,6 +1360,38 @@ def testLegendPlaceholderIcon(self):
...
+        scale_max = 1000
+        rule = QgsRuleBasedRenderer.Rule(symbol, scale_min, scale_max, "fldtxt = 'one'", 'label1', 'description1')
+        rootrule = QgsRuleBasedRenderer.Rule(None)
+        rootrule.appendChild(rule)
+        layer.setRenderer(QgsRuleBasedRenderer(rootrule))
+
+        project.addMapLayers([layer])
+
+        server = QgsServer()
+        request = QgsBufferServerRequest("/?SERVICE=WMS&VERION=1.30&REQUEST=GetLegendGraphic" +
```
Comment (elpaso) [inline tests/src/python/test_qgsserver_wms_getlegendgraphic.py:1382]:
```suggestion
        request = QgsBufferServerRequest("/?SERVICE=WMS&VERSION=1.30&REQUEST=GetLegendGraphic" +
```

**Comment by elpaso** [comment]
@strk is this related to the recent CI changes? Never seen it before: https://github.com/qgis/QGIS/actions/runs/6557297270/job/17814769670?pr=54951#step:14:13

File: tests/src/python/test_qgsserver_wms_getlegendgraphic.py [hunk 2]
```diff
@@ -1360,6 +1360,85 @@ def testLegendPlaceholderIcon(self):
...
+        rule1 = QgsRuleBasedRenderer.Rule(symbol1, scale_min, scale_max, "fldtxt = 'one'", 'label1', 'description1')
+        rootrule.appendChild(rule1)
+        rule2 = QgsRuleBasedRenderer.Rule(symbol2, scale_min, scale_max, "fldtxt = 'two'", 'label2', 'description2')
+        rootrule.appendChild(rule2)
+        layer.setRenderer(QgsRuleBasedRenderer(rootrule))
+
+        project.addMapLayers([layer])
+
+        server = QgsServer()
+        request = QgsBufferServerRequest("/?SERVICE=WMS&VERION=1.30&REQUEST=GetLegendGraphic" +
```
Comment (elpaso) [inline tests/src/python/test_qgsserver_wms_getlegendgraphic.py:1418]:
```suggestion
        request = QgsBufferServerRequest("/?SERVICE=WMS&VERSION=1.30&REQUEST=GetLegendGraphic" +
```

File: tests/src/python/test_qgsserver_wms_getlegendgraphic.py [hunk 3]
```diff
@@ -1360,6 +1360,85 @@ def testLegendPlaceholderIcon(self):
...
+        response = QgsBufferServerResponse()
+        server.handleRequest(request, response, project)
+        j = json.loads(bytes(response.body()))
+        node = j
+        self.assertEqual(node['scaleMaxDenom'], 1000)
+        self.assertEqual(node['scaleMinDenom'], 10000)
+        self.assertEqual(node['rule'], "(fldtxt = 'two') AND (@map_scale <= 1000) AND (@map_scale >= 10000)")
+
+        icon = node['icon']
+        request = QgsBufferServerRequest("/?SERVICE=WMS&VERION=1.30&REQUEST=GetLegendGraphic" +
```
Comment (elpaso) [inline tests/src/python/test_qgsserver_wms_getlegendgraphic.py:1432]:
```suggestion
        request = QgsBufferServerRequest("/?SERVICE=WMS&VERSION=1.30&REQUEST=GetLegendGraphic" +
```

**Comment by elpaso** [comment]
@rldhont @pblottiere can you please review this feature?

**Review by rldhont — APPROVED** [review]

**Comment by elpaso** [comment]
@rldhont thank you!

**Pull request merged**